py_cui/__init__.py

```diff
@@ -1227,34 +1227,43 @@ def move_focus(
 
         self._logger.debug(f"Moved focus to widget {widget.get_title()}")
 
-    def _cycle_widgets(self, reverse: bool = False) -> None:
+    def _cycle_widgets(self, reverse: bool=False, selectable: bool=True) -> None:
         """Function that is fired if cycle key is pressed to move to next widget
 
         Parameters
         ----------
         reverse : bool
             Default false. If true, cycle widgets in reverse order.
+        selectable : bool
+            Default true. If false, non selectable widgets included in cycle.
         """
 
-        num_widgets: int = len(self.get_widgets())
-        current_widget_num: Optional[int] = self._selected_widget
+        select_widget = self.get_widgets()[self._selected_widget]
+        widgets = self.get_widgets()
 
-        if current_widget_num is None:
+        if selectable:
+            selectable = [widgets.get(k) for k in widgets if widgets[k].is_selectable()]
+            widgets = {i: selectable[i] for i in range(0, len(selectable))}
+
+        if select_widget is None:
             return
 
+        get_index = dict((v,k) for k,v in widgets.items())
+        current_widget_num = get_index.get(select_widget)
+
         if reverse:
             next_widget_num = current_widget_num - 1
             if next_widget_num < 0:
-                next_widget_num = num_widgets - 1
+                next_widget_num = len(widgets)
```

Comment:
This should be `len(widgets) - 1` no? Otherwise you'll get an out of bounds error I believe

Reply:
tbh, I haven't experienced that error yet.